src/subscription/data/service.js

```diff
@@ -0,0 +1,50 @@
+import { ensureConfig, getConfig } from '@edx/frontend-platform';
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+
+import handleRequestError from '../../payment/data/handleRequestError';
+import { transformResults } from '../../payment/data/utils';
+
+ensureConfig([
+  'ECOMMERCE_BASE_URL',
+  'LMS_BASE_URL',
+], 'payment API service');
+
+function handleDetailsApiError(requestError) {
+  try {
+    // Always throws an error:
+    handleRequestError(requestError);
+  } catch (errorWithMessages) {
+    const processedError = new Error();
+    processedError.messages = errorWithMessages.messages;
+    processedError.errors = errorWithMessages.errors;
+    processedError.fieldErrors = errorWithMessages.fieldErrors;
+
+    if (requestError.response.data) {
+      processedError.details = transformResults(requestError.response.data);
+    }
+
+    throw processedError;
+  }
+}
+
+export async function getCaptureKey() {
+  const { data } = await getAuthenticatedHttpClient()
+    .get(`${getConfig().ECOMMERCE_BASE_URL}/bff/payment/v0/capture-context/`)
+    .catch(handleDetailsApiError);
+  return data;
+}
+
+export async function getClientSecret() {
+  const { data } = await getAuthenticatedHttpClient()
+    .get(`${getConfig().ECOMMERCE_BASE_URL}/bff/payment/v0/capture-context`)
+    .catch(handleDetailsApiError);
+  return data;
+}
+
+export async function getDetails(discountJwt) {
+  const discountJwtArg = typeof discountJwt !== 'undefined' ? `?discount_jwt=${discountJwt}` : '';
+  const { data } = await getAuthenticatedHttpClient()
+    .get(`${getConfig().ECOMMERCE_BASE_URL}/bff/payment/v0/payment/${discountJwtArg}`)
```

Comment:
Can you add a TODO comment here as well to replace with correct URL?

Reply:
Done.